presto-native-execution/presto_cpp/main/QueryContextManager.cpp

```diff
@@ -98,15 +98,26 @@ void updateVeloxConnectorConfigs(
     std::unordered_map<
         std::string,
         std::unordered_map<std::string, std::string>>& connectorConfigStrings) {
+  const auto& systemConfig = SystemConfig::instance();
+
   for (auto& entry : connectorConfigStrings) {
     auto& connectorConfig = entry.second;
 
-    // Do not retain cache if `node_selection_strategy` is explicitly set to
-    // `NO_PREFERENCE`.
     auto it = connectorConfig.find("node_selection_strategy");
-    if (it != connectorConfig.end() && it->second == "NO_PREFERENCE") {
-      connectorConfig.emplace(
-          connector::hive::HiveConfig::kCacheNoRetentionSession, "true");
+    if (systemConfig->asyncCacheSsdByDefaultEnabled()) {
+      // If SSD cache is enabled by default, do not retain cache if
+      // `node_selection_strategy` is explicitly set to `NO_PREFERENCE`.
+      if (it != connectorConfig.end() && it->second == "NO_PREFERENCE") {
+        connectorConfig.emplace(
+            connector::hive::HiveConfig::kCacheNoRetentionSession, "true");
+      }
+    } else {
+      // If SSD cache is disabled by default, only retain cache if
+      // `node_selection_strategy` is explicitly set to `SOFT_AFFINITY`.
+      if (it == connectorConfig.end() || it->second == "SOFT_AFFINITY") {
```

Comment:
shouldn't this be `it == connectorConfig.end() || it->second == "NO_PREFERENCE"`?